doc/developer/design/20230225_compute_introspection_schema.md

```diff
@@ -0,0 +1,217 @@
+reporting `environmentd`'s. For example, `mz_compute_frontiers`
+vs. `mz_cluster_replica_frontiers`. However, the names do not make it
+clear what the distinction is. 
+
+### Suggested fixes for schema inconsistencies
+
+* All per-worker relations should have a corresponding "global" view
+  where the results are aggregated across workers in the way most
+  likely to be useful (for example, summing, taking the minimum, or
+  restricting to worker 0). Because these global views are more likely
+  to be useful than the underlying sources, they should have the basic
+  name (e.g., `mz_arrangement_sharing`). The underlying source should
+  have a name with `per_worker` suffixed (e.g.,
+  `mz_arrangement_sharing_per_worker`). Furthermore, the per-worker
+  variant should be clearly less prominent in documentation (either at
+  the bottom of the page, or in a "see more" section that has to be
+  clicked to expand).
+* The expectations about per-worker relations should be clearly
+  documented (e.g., we should clearly say whether the values for each
+  worker are expected to be the same during normal operation).
+* All relations that report a simple count or sum should be views based on
+  an `_internal` variant that maintains the count or sum in the
+  Differential diff field. We may consider renaming this, since it
+  conflicts confusingly with the overall name `mz_internal`.
+* `raw` variants should be removed, and the underlying dataflows
+  should be in terms of intervals.
+* All histogram views or sources should have a per-bucket `sum`
+  column.
+* All histogram views or sources should be suffixed with `_histogram`,
+  to make their role more obvious. 
+* All per-replica sources and views should have their names prepended
+  with some string (perhaps `clusterd_`, though I'm open to
+  suggestions) that makes it obvious which ones they are. Furthermore,
+  the limitations of per-replica relations should be clearly
+  documented.
+
+### Possibly useful missing views
+
+#### Channel-to-operator mapping
+
+See [GitHub
+PR](https://github.com/MaterializeInc/materialize/pull/17825).
+
+#### Scope child to parent mapping
+
+The only way to understand the structure of scope nesting currently is
+to use the `mz_dataflow_addresses` relation and do some tedious list
+manipulation. A simple view of `(id, parent_id)` would make this much
+simpler.
+
+#### Transitive object dependencies and end-to-end lag
+
+We should add the transitive closure of `mz_object_dependencies`,
+called e.g. `mz_transitive_object_dependencies`.
+
+This will allow us to see the end-to-end frontier lag for any
+dataflow.
+
+#### More views enriched with operator names, dataflow names, etc.
+
+(Not quite sure about this -- it's tricky to balance ease-of-use with
+not blowing up the set of documented relations)
+
+## Alternatives
+
+None known to me
+
+## Open questions
+
+* Are there further views that could be useful but that we don't have
```

Comment:
When calculating frontier lag, we usually have to write some tedious joins between `mz_compute_frontiers` and `mz_compute_import_frontiers`. Perhaps a ready-made view there would be pretty useful?